code/model/SiteTree.php

```diff
@@ -439,7 +439,6 @@ static public function link_shortcode_handler($arguments, $content = null, $pars
 		if (
 			   !($page = DataObject::get_by_id('SiteTree', $arguments['id']))         // Get the current page by ID.
 			&& !($page = Versioned::get_latest_version('SiteTree', $arguments['id'])) // Attempt link to old version.
-			&& !($page = DataObject::get_one('ErrorPage', array('"ErrorPage"."ErrorCode"' => 404)))   // Link to 404 page directly.
```

Comment:
I guess I could leave this in (and related tests) by testing for existence of ErrorPage class, but since this covers an edge case which shouldn't occur and is not a great fallback anyway (who wants to click a link to a 404 page) I think it would be okay if we don't render anything in the case of an ID not pointing to any valid page ID.